SavedInstances/Modules/Progress.lua

```diff
@@ -310,6 +310,44 @@ local function DragonflightRenownReset(toon, index)
   -- do nothing
 end
 
+-- Primal Storms Core
+local function PrimalStormsCoreUpdate(index)
+  SI.db.Toons[SI.thisToon].Progress[index] = wipe(SI.db.Toons[SI.thisToon].Progress[index] or {})
+  for _, questID in ipairs(Module.TrackedQuest[index].relatedQuest) do
+    SI.db.Toons[SI.thisToon].Progress[index][questID] = C_QuestLog_IsQuestFlaggedCompleted(questID)
+  end
+end
+
+local function PrimalStormsCoreShow(toon, index)
+  local t = SI.db.Toons[toon]
+  if not t or not t.Quests then return end
+  if not t or not t.Progress or not t.Progress[index] then return end
+
+  local stringTypeCore = {
+    YELLOW_FONT_COLOR_CODE .. "E" .. FONT_COLOR_CODE_CLOSE,
+    "|cff42a4f5" .. "W" .. FONT_COLOR_CODE_CLOSE,
+    "|cffe4f2f5" .. "A" .. FONT_COLOR_CODE_CLOSE,
+    ORANGE_FONT_COLOR_CODE .. "F" .. FONT_COLOR_CODE_CLOSE
+  }
+  local finalString = ""
+  for i, questID in ipairs(Module.TrackedQuest[index].relatedQuest) do
+    if t.Progress[index][questID] then
+      if finalString ~= "" then
+        finalString = finalString .. ' / '
+      end
+      finalString = finalString .. stringTypeCore[i]
+    end
+  end
+  return finalString == "" and "None" or finalString
+end
```

Comment:
I'd prefer use number progress like `1/4` as first letter abbr is hard to localize and understand in other languages.